targets/nuttx-stm32f4/jerry_main.c

```diff
@@ -51,126 +52,66 @@ print_help (char *name)
  */
-static char*
-read_sources (const char *script_file_names[],
-              int files_count,
-              size_t *out_source_size_p)
+static const uint8_t *
+read_file (const char *file_name, /**< source code */
+           size_t *out_size_p) /**< number of bytes successfully read from source */
 {
-  int i;
-  char* source_buffer = NULL;
-  char *source_buffer_tail = NULL;
-  size_t total_length = 0;
-  FILE *file = NULL;
-
-  for (i = 0; i < files_count; i++)
+  FILE *file = fopen (file_name, "r");
+  if (file == NULL)
   {
-    const char *script_file_name = script_file_names[i];
-
-    file = fopen (script_file_name, "r");
-    if (file == NULL)
-    {
-      jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Failed to fopen [%s]\n", script_file_name);
-      return NULL;
-    }
-
-    int fseek_status = fseek (file, 0, SEEK_END);
-    if (fseek_status != 0)
-    {
-      jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Failed to fseek fseek_status(%d)\n", fseek_status);
-      fclose (file);
-      return NULL;
-    }
-
-    long script_len = ftell (file);
-    if (script_len < 0)
-    {
-      jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Failed to ftell script_len(%ld)\n", script_len);
-      fclose (file);
-      break;
-    }
-
-    total_length += (size_t) script_len;
-
-    fclose (file);
-    file = NULL;
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: failed to open file: %s\n", file_name);
+    return NULL;
   }
 
-  if (total_length <= 0)
+  int fseek_status = fseek (file, 0, SEEK_END);
+  if (fseek_status != 0)
   {
-    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "There's nothing to read\n");
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Failed to fseek fseek_status(%d)\n", fseek_status);
+    fclose (file);
     return NULL;
   }
 
-  source_buffer = (char*) malloc (total_length);
-  if (source_buffer == NULL)
+  long script_len = ftell (file);
+  if (script_len < 0)
   {
-    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Out of memory error\n");
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Failed to ftell script_len(%ld)\n", script_len);
+    fclose (file);
     return NULL;
   }
-  memset (source_buffer, 0, sizeof (char) * total_length);
-  source_buffer_tail = source_buffer;
-
-  for (i = 0; i < files_count; i++)
-  {
-    const char *script_file_name = script_file_names[i];
-    file = fopen (script_file_name, "r");
-
-    if (file == NULL)
-    {
-      jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Failed to fopen [%s]\n", script_file_name);
-      break;
-    }
-
-    int fseek_status = fseek (file, 0, SEEK_END);
-    if (fseek_status != 0)
-    {
-      jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Failed to fseek fseek_status(%d)\n", fseek_status);
-      break;
-    }
 
-    long script_len = ftell (file);
-    if (script_len < 0)
-    {
-      jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Failed to ftell script_len(%ld)\n", script_len);
-      break;
-    }
+  rewind (file);
 
-    rewind (file);
+  uint8_t *buffer = jmem_heap_alloc_block (script_len);
```

Comment:
void *jmem_heap_alloc_block (const size_t size);
    void *jmem_heap_alloc_block_null_on_error (const size_t size);

The jmem_heap_alloc_block crashes on error.